internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubMessage.java

```
@@ -39,6 +40,20 @@ default POSITION getOffset() {
    */
   int getPayloadSize();
 
+  /**
+   * @return the schema id that the record was written with
+   */
+  default int getWriterSchemaId() {
+    throw new VeniceException("WriterSchemaId is not available from the source this message was created");
```

Any reason why throw exception is a better default behavior then simply return `-1` and `null` for these new metadata fields in the `PubSubMessage`?

---

If someone tries to call `getWriterSchemaId` in a new code path internally, I think failing loudly than silently is better. It will make debugging easier, if someone decides to use it.